### ISSUE TITLE
Provider fails to remove VCS configuration 

### ISSUE THREAD

**Issue by peimanja** [terraform-providers/terraform-provider-tfe#91]
Using `tfe_workspace` resource and it successfully creates the workspaces and VCS configuration:
```hcl
resource "tfe_workspace" "test" {
  name              = "test"
  organization      = "my-organization"
  auto_apply        = true
  terraform_version = "0.12.7"
  working_directory = "terraform"

  vcs_repo {
    identifier     = "my-organization/my-repo"
    oauth_token_id = "********"
  }
}
```
everything looks fine in . TFE, but if I remove the VCS config after, Terraform detects that it needs to remove it and it applies without any error but it does not remove the VCS config and the change shows up . in every run until the VCS config manually get removed.

```
Terraform v0.12.7
+ provider.tfe v0.11.0```

**Comment by gfysaris** [comment]
+1 
I have the exact same problem.

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # module.tfe_organizations.module.evt_organization.module.tfe_workspaces.module.XXXXX.tfe_workspace.ws[0] will be updated in-place
  ~ resource "tfe_workspace" "ws" {
        auto_apply            = false
        external_id           = "ws-XXXXX"
        file_triggers_enabled = true
        id                    = "XXX/XXXX"
        name                  = "XXXX"
        operations            = true
        organization          = "XXX"
        queue_all_runs        = true
        terraform_version     = "0.12.12"
        trigger_prefixes      = []

      - vcs_repo {
          - branch             = "master" -> null
          - identifier         = "XXX/XXXX" -> null
          - ingress_submodules = true -> null
          - oauth_token_id     = "ot-XXXXX" -> null
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions in workspace "XXXXX"?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.tfe_organizations.module.evt_organization.module.tfe_workspaces.module.XXXXX.tfe_workspace.ws[0]: Modifying... [id=XXX/XXXX]
module.tfe_organizations.module.evt_organization.module.tfe_workspaces.module.XXXXX.tfe_workspace.ws[0]: Modifications complete after 1s [id=XXX/XXXX]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

but then when i check the VCS config for the workspace (either from UI or API)

```
        {
            "id": "ws-XXXXX",
            "type": "workspaces",
            "attributes": {
                "name": "XXXXX",
                "auto-apply": false,
                "created-at": "2019-10-25T08:22:30.971Z",
                "environment": "default",
                "locked": false,
                "queue-all-runs": true,
                "terraform-version": "0.12.12",
                "working-directory": null,
                "speculative-enabled": true,
                "latest-change-at": "2019-10-28T19:15:42.704Z",
                "operations": true,
                "vcs-repo": {
                    "branch": "master",
                    "ingress-submodules": true,
                    "identifier": "XXX/XXXX",
                    "display-identifier": "XXX/XXXX",
                    "oauth-token-id": "ot-XXXXX",
                    "webhook-url": "https://app.terraform.io/webhooks/vcs/XXXXX-XXXX-XXXX-XXXX-XXXXX"
                },
                "permissions": {
                    "can-update": false,
                    "can-destroy": false,
                    "can-queue-destroy": true,
                    "can-queue-run": true,
                    "can-queue-apply": true,
                    "can-update-variable": true,
                    "can-lock": true,
                    "can-unlock": true,
                    "can-force-unlock": false,
                    "can-read-settings": true
                },
                "actions": {
                    "is-destroyable": true
                },
                "description": null,
                "file-triggers-enabled": true,
                "trigger-prefixes": [],
                "source": "tfe-api",
                "source-name": null,
                "source-url": null
            },
            "relationships": {
                "organization": {
                    "data": {
                        "id": "XXXX",
                        "type": "organizations"
                    }
                },
                "current-run": {
                    "data": {
                        "id": "run-7ECF2TFbqFWhWgSs",
                        "type": "runs"
                    },
                    "links": {
                        "related": "/api/v2/runs/run-7ECF2TFbqFWhWgSs"
                    }
                },
                "latest-run": {
                    "data": {
                        "id": "run-7ECF2TFbqFWhWgSs",
                        "type": "runs"
                    },
                    "links": {
                        "related": "/api/v2/runs/run-7ECF2TFbqFWhWgSs"
                    }
                },
                "current-state-version": {
                    "data": {
                        "id": "sv-gMwCZzfRJFxYBu9Y",
                        "type": "state-versions"
                    },
                    "links": {
                        "related": "/api/v2/workspaces/ws-XXXXX/current-state-version"
                    }
                }
            },
            "links": {
                "self": "/api/v2/organizations/XXX/workspaces/XXXX"
            }
        },
```

**Comment by mkjmdski** [comment]
in 0.15.0 with terraform 0.12.24 this problem still exists but seems more like too fast API consumption as it occurs when changing VCS connection ID and fails on the first run when terraform tries to replace the id by removal and addition.

**Comment by peimanja** [comment]
Yep it is still an issue

**Comment by lafentres** [comment]
A fix for this issue is coming soon in #173

**Comment by lafentres** [comment]
Just released version 0.16.2, which should fix this issue! 🎉
